HLTrigger/Timer/interface/FastTimerService.h

```diff
@@ -86,690 +82,488 @@ class FastTimerService {
   FastTimerService(const edm::ParameterSet &, edm::ActivityRegistry & );
   ~FastTimerService();
 
-  // reserve plots for timing vs. luminosity - these should only be called before any begin run (actually, before any preStreamBeginRun)
-  unsigned int reserveLuminosityPlots(std::string const & name, std::string const & title, std::string const & label, double range, double resolution);
-  unsigned int reserveLuminosityPlots(std::string && name, std::string && title, std::string && label, double range, double resolution);
-
-  // set the event luminosity
-  void setLuminosity(unsigned int stream_id, unsigned int luminosity_id, double value);
-
-  // query the current module/path/event
-  // Note: these functions incur in a "per-call timer overhead" (see above), currently of the order of 340ns
-  double currentModuleTime(edm::StreamID) const;            // return the time spent since the last preModuleEvent() event
-  double currentPathTime(edm::StreamID) const;              // return the time spent since the last prePathEvent() event
-  double currentEventTime(edm::StreamID) const;             // return the time spent since the last preEvent() event
+private:
+  double queryModuleTime_(edm::StreamID, unsigned int id) const;
 
-  // query the time spent in a module/path (available after it has run)
-  double queryModuleTime(edm::StreamID, const edm::ModuleDescription &) const;
-  double queryModuleTime(edm::StreamID, unsigned int id) const;
-  double queryModuleTimeByLabel(edm::StreamID, const std::string &) const;
-  double queryModuleTimeByType(edm::StreamID, const std::string &) const;
-  /* FIXME re-implement taking into account subprocesses
-  double queryPathActiveTime(edm::StreamID, const std::string &) const;
-  double queryPathExclusiveTime(edm::StreamID, const std::string &) const;
-  double queryPathTotalTime(edm::StreamID, const std::string &) const;
-  */
-
-  // query the time spent in the current event's
-  //  - source        (available during event processing)
-  //  - all paths     (available during endpaths)
-  //  - all endpaths  (available after all endpaths have run)
-  //  - processing    (available after the event has been processed)
+public:
+  // query the time spent in a module/path/process (available after it has run)
```

Comment:
So, based on your comment and assuming that the query is for a module in the same stream
  - if the module has already run, it was on a different task, so they should be safe because of TBB synchronisation;
 - if the module is running or has not yet run, query should return 0, because the value is store only after the end of the module.

Anyway, I am not very fond of the query functions myself - eventually I would like to make the information for all Paths available in the event (in a similar way that the TriggerResults are made available).